### PR TITLE
📆 feat(components): add native date range filter input

### PR DESCRIPTION
**Pull request by adriangohjw**
### 📊 PR diff breakdown

| Bucket | Files | +Lines | -Lines |
|---|---|---|---|
| ⚙️ App | 2 | +168 | -0 |
| 🧪 Test | 2 | +187 | -0 |
| 📄 Doc | 1 | +31 | -0 |
| 🚫 Ignore | 0 | +0 | -0 |

## Problem

Collection date filters need a date range picker UI so visitors can select start/end dates when filtering by date-type tag categories.

Replaces #3300 (auto-merged during an incorrect rebase).

Depends on #3299

## Solution

Replace the earlier react-aria calendar implementation with two native `<input type="date">` fields ("From" / "To"). This removes calendar-only dependencies from the published components package while keeping the filter input ready for wiring into `Filter` / `FilterDrawer`.

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Features**:

- Add `DateRangeFilterInput` with side-by-side native date inputs for From and To
- Support open-ended ranges (`{ start }`, `{ end }`, or both)
- Live `onChange` with inline validation when From is after To
- Add Storybook stories and browser tests for empty, closed, and open-ended states

**Improvements**:

- Remove custom `RangeCalendar`, masked text input, and date parsing/formatting utilities (~1.2k lines)
- Remove `@react-aria/calendar`, `@react-stately/calendar`, and `@internationalized/date` from `packages/components`

## Before & After Screenshots

**BEFORE**:

**AFTER**:

## Tests

**Manual Verification Steps**:

- [ ] Open Storybook and review `DateRangeFilterInput` stories (empty, closed range, From-only, To-only)
- [ ] Set a closed range and confirm `onChange` receives `{ start, end }` in ISO `yyyy-MM-dd` format
- [ ] Set only From or only To and confirm open-ended values pass through
- [ ] Set From after To and confirm an inline validation error appears without committing the invalid range
- [ ] Clear one field independently and confirm the other value is preserved
- [ ] Run `pnpm test:unit -- src/templates/next/components/internal/Filter/DateRangeFilterInput/__tests__/DateRangeFilterInput.browser.test.tsx` in `packages/components`
- [ ] Run `pnpm typecheck` in `packages/components`

**Removed dependencies**:

- `@internationalized/date`, `@react-aria/calendar`, `@react-stately/calendar`

## /show-me to help explain what this PR does

New self-contained component that turns two native `<input type="date">` fields into one `DateRangeFilterValue`:

```text
packages/components/src/templates/next/components/internal/Filter/DateRangeFilterInput/
├── DateRangeFilterInput.tsx           # the component (166 lines)
├── DateRangeFilterInput.stories.tsx   # Empty / WithRangeApplied / WithFromOnly / WithToOnly
├── index.ts                           # exports DateRangeFilterInput + DateRangeFilterValue
└── __tests__/
    └── DateRangeFilterInput.browser.test.tsx
docs/adr/0009-native-date-inputs-for-date-range-filter.md   # why native inputs, not a react-aria calendar
```

```mermaid
flowchart LR
    subgraph DateRangeFilterInput
        From["input#From type=date\nmax={end}"]
        To["input#To type=date\nmin={start}"]
        commit["commitIfValid()"]
    end
    From -- handleStartChange --> commit
    To -- handleEndChange --> commit
    commit -- "start > end?" --> Error["setValidationError(true)\nrender role=alert"]
    commit -- valid --> Filter["toFilterValue(start, end)"]
    Filter -- "onChange({start?, end?} | undefined)" --> Parent[/consumer, e.g. Filter/FilterDrawer/]
```

Each field constrains the other natively (`max={end}` / `min={start}`), so the browser blocks most invalid picks up front; `commitIfValid` is the fallback for what native constraints don't cover (typed/pasted values) and only calls `onChange` once the range is valid. `undefined` is emitted when both fields are empty, and an open-ended range (`{ start }` or `{ end }` only) is a valid committed value.

The last commit fixes one interaction bug: dismissing the native date picker popup with Escape was also closing the parent mobile filter dialog.

```diff
 <input
   type="date"
   onChange={handleStartChange}
+  onKeyDown={stopEscapeFromBubbling}
 />
```

```ts
// Native date pickers close on Escape. Stop bubbling so a parent dialog
// (the mobile filter drawer) does not close at the same time.
const stopEscapeFromBubbling = (event: React.KeyboardEvent) => {
  if (event.key === "Escape") {
    event.stopPropagation()
  }
}
```

<h2><a href="https://app.greptile.com/api/retrigger?id=59848103"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/RetriggerDark.svg?v=2"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/Retrigger.svg?v=2">Retrigger</picture></a>Confidence Score: 5/5</h2>

The PR appears safe to merge because no blocking failure remains.

<details><summary><h3>Summary</h3></summary>

The PR replaces the earlier custom calendar approach with a self-contained native date-range input for collection filters.
- Adds controlled From and To date fields supporting closed and open-ended ranges.
- Prevents invalid ranges from being committed and displays an inline validation error.
- Stops Escape from bubbling to the parent mobile filter dialog.
- Adds Storybook states, browser interaction tests, local exports, and an ADR documenting the native-input tradeoffs.
</details>

<details open><summary><h3>Diagram</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  From[From date input] --> Validate[Validate range]
  To[To date input] --> Validate
  Validate -->|Start is after end| Error[Show inline error]
  Validate -->|Valid closed or open range| Change[Call onChange]
  Change --> Parent[Parent filter state]
```
</details>

<sub>Reviews (3) · Last reviewed commit: ["feat(components): prevent dialog closure..."](https://github.com/opengovsg/isomer/commit/81755d69ec6e2d48773dfc9610d5a66b5e6f55f8)</sub>

### CODE REVIEW

File: packages/components/src/templates/next/components/internal/Filter/DateRangeFilterInput.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
+"use client"
+
+import { Dialog, DialogBackdrop, DialogPanel } from "@headlessui/react"
+import { parseDate } from "@internationalized/date"
+import { useEffect, useRef, useState } from "react"
+import { BiCalendar } from "react-icons/bi"
+import { useOnClickOutside } from "usehooks-ts"
+import { DEFAULT_DATE_RANGE_FILTER_LABEL } from "~/types/constants"
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/templates/next/components/internal/Filter/DateRangeFilterInput.tsx]: Fixed in aa33ec698 — removed the `DEFAULT_DATE_RANGE_FILTER_LABEL` import (the constant was never exported on this branch after the card-header rebase) and inlined the label text in the component. `pnpm exec tsc --noEmit` in `packages/components` passes.

File: packages/components/src/templates/next/components/internal/Filter/RangeCalendar.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+import {
+  useCalendarCell,
+  useCalendarGrid,
+  useRangeCalendar,
+} from "@react-aria/calendar"
+import { useRangeCalendarState } from "@react-stately/calendar"
+import { useMemo, useRef, useState } from "react"
+import { BiChevronLeft, BiChevronRight } from "react-icons/bi"
+
+import { getTodayInSingapore } from "../../../layouts/Collection/utils/getDateFilterStatus"
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/templates/next/components/internal/Filter/RangeCalendar.tsx]: Fixed in aa33ec698 — replaced the broken `getTodayInSingapore` import from `layouts/Collection/utils/getDateFilterStatus` (removed during the card-header rebase) with `getSingaporeDateYYYYMMDD` from `~/utils/getSingaporeDate`.

File: packages/components/src/templates/next/components/internal/Filter/DateRangeFilterInput/DateRangeFilterInput.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
+"use client"
+
+import { Dialog, DialogBackdrop, DialogPanel } from "@headlessui/react"
+import { parseDate } from "@internationalized/date"
+import { useEffect, useRef, useState } from "react"
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/templates/next/components/internal/Filter/DateRangeFilterInput/DateRangeFilterInput.tsx]: Keeping `"use client"` here — this component owns popover/modal state, typed input validation, and calendar interaction, same as the existing Filter subsystem (`Filter.tsx`, `FilterDrawer.tsx`, `RangeCalendar.tsx` are all client components). A server wrapper would not remove the hydration boundary since the parent filter sidebar is already client-rendered.

**Review by Ssatu — APPROVED** [review]
